lib/PuppeteerSharp/Connection.cs

```diff
@@ -184,7 +183,7 @@ internal void Close(string closeReason)
         #region Private Methods
 
         private async void Transport_MessageReceived(object sender, MessageReceivedEventArgs e)
-            => await _callbackQueue.Enqueue(() => ProcessMessage(e)).ConfigureAwait(false);
+            => await ProcessMessage(e).ConfigureAwait(false);
```

Comment:
This was necessary to prevent marshalling WS messages one by one; otherwise, my example would still have a deadlock while waiting for the semaphore before handling the next message.

Reply:
Found that this change wasn't necessary, and caused Issue0716 test to fail. Reverted.